space_packet_parser/packets.py

```diff
@@ -56,28 +73,87 @@ class StrParameter(_Parameter, str):
 ParameterDataTypes = Union[BinaryParameter, BoolParameter, FloatParameter, IntParameter, StrParameter]
 
 
+class SequenceFlags(IntEnum):
+    """Enumeration of the possible sequence flags in a CCSDS packet."""
+    CONTINUATION = 0
+    FIRST = 1
+    LAST = 2
+    UNSEGMENTED = 3
+
+
 class RawPacketData(bytes):
     """A class to represent raw packet data as bytes but whose length is represented by bit length.
 
     This class is a subclass of bytes and is used to represent the raw packet data
     in a more readable way. It is used to store the raw packet data in the Packet
-    class and used to keep track of the current parsing position.
+    class and used to keep track of the current parsing position (accessible through the `pos` attribute).
 
-    Parameters
-    ----------
-    data : bytes
-        Raw packet data. Full CCSDS packets are always an integer number of bytes.
+    There are also a few convenience methods to extract the CCSDS header fields from the packet data.
     """
-    def __init__(self, data: bytes, *, pos: int = 0):
-        self.pos = pos
-        self._nbits = len(data) * 8
-        super().__init__()
+    HEADER_LENGTH_BYTES = 6
+    pos = 0  # in bits
+
+    def __str__(self) -> str:
+        return (f"RawPacketData Header: ({self.version_number=}, {self.type=}, "
+                f"{self.secondary_header_flag=}, {self.apid=}, {self.sequence_flags=}, "
+                f"{self.sequence_count=}, {self.data_length=})").replace("self.", "")
+
+    @cached_property
+    def version_number(self) -> int:
+        """CCSDS Packet Version Number"""
+        return _extract_bits(self, 0, 3)
+
+    @cached_property
+    def type(self) -> int:
```

Comment:
`type` is a built-in function in Python, so do we care that this is called `RawPacketData.type`? Should we prefix it with something like `packet_type` instead? To me that is redundant, but happy to change if others want a different naming.